cache-manager.php

```diff
@@ -246,11 +248,65 @@ function queue_post_purge( $post_id ) {
 		}
 
 		$this->purge_urls[] = get_permalink( $post_id );
+	}
+	
+	function transition_post_status( $new, $old, $post ) {
+		if ( $this->site_cache_purged ) {
+			return;
+		}
+		
+		if ( defined( 'WP_IMPORTING' ) ) {
+			return;
+		}
+		
+		// Only send PURGE requests for public post types
+		if ( ! is_post_type_viewable( $post->post_type ) ) {
+			return;
+		}
+		
+		// Don't PURGE if we're not transitioning to or from publish
+		if ( $new === $old || ( $new != 'publish' && $old != 'publish' ) ) {
+			return;
+		}
+		
+		$this->clear_feed_urls( $post );
+	}
+
+	function post_updated( $post_id, $post, $old_post ) {
+		if ( $this->site_cache_purged ) {
+			return;
+		}
+
+		if ( defined( 'WP_IMPORTING' ) ) {
+			return;
+		}
+
+		// Only send PURGE requests for public post types
+		if ( ! is_post_type_viewable( $post->post_type ) ) {
+			return;
+		}
+
+		if ( 'publish' !== get_post_status( $post_id ) ) {
+			return
+		}
+
+		$fields = apply_filters( 'wpcom_vip_cache_purge_on_updated_fields', array( 'post_title', 'post_excerpt' ), $post );
+
+		// PURGE all feeds if one of the whitelisted fields is updated
+		foreach( $fields as $field ) {
+			if ( $post->{$field} != $old_post->{$field} ) {
+				$this->clear_feed_urls( $post );
+				return;
+			}
+		}
+	}
+
+	function clear_feed_urls( $post ) {
```

Comment:
We may want to name this function differently, since it purges more than just feeds.